#! /bin/sh /usr/share/dpatch/dpatch-run
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Change --version to report that this is part of Puppet
Enterprise,
## DP: and which version of Puppet Enterprise it is part of.

@DPATCH@


From f71e67957afdf795bf282f4faf89e188358b9b0f Mon Sep 17 00:00:00 2001
From: Nigel Kersten <nigel@puppetlabs.com>
Date: Thu, 3 Nov 2011 18:32:56 -0700
Subject: [PATCH] (#10543) Rake task to add all nodes to a group

---
 lib/tasks/classesgroups.rake |   29 +++++++++++++++++++++++++++++
 1 files changed, 29 insertions(+), 0 deletions(-)

diff --git a/lib/tasks/classesgroups.rake b/lib/tasks/classesgroups.rake
index d4b18fc..b15eb94 100644
--- a/lib/tasks/classesgroups.rake
+++ b/lib/tasks/classesgroups.rake
@@ -217,4 +217,33 @@ namespace :nodegroup do
       puts e.message
     end
   end
+
+  desc 'Automatically adds all nodes to a group'
+  task :add_all_nodes => :environment do
+    if ENV['group']
+      groupname = ENV['group']
+    else
+      puts 'Must specify group name (group=<groupname>).'
+      exit 1
+    end
+
+    group = NodeGroup.find_by_name(groupname)
+    if group.nil?
+      puts "Cannot find group: #{groupname}"
+      exit 1
+    end
+
+    begin
+      Node.find(:all).each do |node|
+        node_groups = node.node_groups
+        unless node_groups.include?(group)
+          node_groups.push(group)
+          node.node_groups = node_groups
+        end
+      end
+    rescue => e
+      puts "There was a problem adding all nodes to the group '#{group}': #{e.message}"
+      exit 1
+    end
+  end
 end
-- 
1.7.6

